olive/engine/engine.py

```diff
@@ -111,22 +112,21 @@ def __init__(
                 accelerators = inferred_accelerators
 
         not_supported_ep = set()
-        processed_ep = set()
         self.accelerator_specs: List[AcceleratorSpec] = []
         is_cpu_available = "cpu" in [accelerator.lower() for accelerator in accelerators]
         for accelerator in accelerators:
             device = Device(accelerator.lower())
-            supported_eps = AcceleratorLookup.get_execution_providers_for_device(device)
-            eps = [e for e in self.execution_providers if e not in processed_ep]
-            for ep in eps:
+            skip_get_available_ep = False
+            if self.target.system_type in (SystemType.AzureML, SystemType.Docker, SystemType.PythonEnvironment):
+                skip_get_available_ep = True
+            supported_eps = AcceleratorLookup.get_execution_providers_for_device(device, skip_get_available_ep)
+            for ep in self.execution_providers:
                 if ep not in supported_eps:
                     not_supported_ep.add(ep)
-                    processed_ep.add(ep)
                 elif ep == "CPUExecutionProvider" and device != "cpu" and is_cpu_available:
                     logger.info("ignore the CPUExecutionProvider for non-cpu device")
                 else:
                     self.accelerator_specs.append(AcceleratorSpec(device, ep))
-                    processed_ep.add(ep)
 
```

Comment:
If there are two accelerators(for e.g. GPU and CPU), original logic will only try CPU EP once. The new behavior will run twice?

Reply:
we want to try CPU and GPU if user wants to run openvino and cuda execution provider.